x/oracle/types/prophecy.go

```diff
@@ -0,0 +1,144 @@
+package types
+
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/x/staking"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const PendingStatusText = "pending"
+const SuccessStatusText = "success"
+const FailedStatusText = "failed"
+
+// Prophecy is a struct that contains all the metadata of an oracle ritual.
+// Claims are indexed by the claim's validator bech32 address and by the claim's json value to allow
+// for constant lookup times for any validation/verifiation checks of duplicate claims
+// Each transaction, pending potential results are also calculated, stored and indexed by their byte result
+// to allow discovery of consensus on any the result in constant time without having to sort or run
+// through the list of claims to find the one with highest consensus
+
+type Prophecy struct {
+	ID              string                      `json:"id"`
+	Status          Status                      `json:"status"`
+	ClaimValidators map[string][]sdk.ValAddress `json:"claim_validators"` //This is a mapping from a claim to the list of validators that made that claim

```

Comment:
Is this an issue, or just a reminder?

As far as i can tell, the way each client stores the map in the DB will be nondeterministic, but because we only access data from the map in a deterministic way, there is no risk of nondeterminism at consensus level, right? I guess if there was any kind of iteration over that mapping during the business logic/checks that cared about the order of iteration, then that would cause consensus failure?

Reply:
Hmm what about using `[][]byte`? I'm not entirely sure how gas works in the SDK, but if it is used during iteration then it is possible a consensus failure could occur if a transaction had a gas limit borderline enough to fail on some nodes and succeed on others

Reply:
The reason for using a mapping is for performant o(1) queries when doing checks. I don't think a 2d byte array will be as performant, and that would also then need a mapping from claim id to validator array? 

It is not used and should not be used in iteration. Will add an additional comment to warn against that.